src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx

```diff
@@ -119,7 +119,7 @@ const TimeEntryForm = props => {
   const [projectsAndTasksOptions, setProjectsAndTasksOptions] = useState([]);
   const [submitting, setSubmitting] = useState(false);
 
-  const canEditTimeEntry = props.hasPermission('editTimelogInfo') || props.hasPermission('editTimeEntry');
+  const canEditTimeEntry = props.hasPermission('editTimelogInfo') || props.hasPermission('editTimeEntry') || props.hasPermission('editTimelogDate');
```

Comment:
This should be separated so that `editTimelogDate` only gives access to change the date and no other fields.

Reply:
Also needs a backend PR.

Reply:
Done. Please let me know if other changes are needed.